### PR TITLE
Input: use mousedown to fix select blur event

### LINKED ISSUE #10810: [Bug Report] <el-select> unexpected blur event
### Element UI version
2.3.5

### OS/Browsers version
win7 / chrome 66

### Vue version
2.5.16

### Reproduction Link
[https://jsfiddle.net/mmx38qxw/1840/](https://jsfiddle.net/mmx38qxw/1840/)

### Steps to reproduce
- select an item
- "@blur" should not be triggered

### What is Expected?
no blur event when a value is select

### What is actually happening?
blur event is triggered when an option is selected

### LINKED ISSUE #13067: [Bug Report] el-select trigger blur when call focus()
### Element UI version
2.4.8

### OS/Browsers version
mac/chrome

### Vue version
2.5.17

### Reproduction Link
[https://jsfiddle.net/a1L3vygb/1/](https://jsfiddle.net/a1L3vygb/1/)

### Steps to reproduce
programmatic trigger `el-select` focus() method

### What is Expected?
focus in el-select, not trigger blur event

### What is actually happening?
will trigger once blur event

### PR DESCRIPTION
**Pull request by island205**
- Closes #10810
- Closes #13067

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: packages/select/src/select.vue [hunk 1]
```diff
@@ -661,7 +659,8 @@
         }, 300);
       },
 
-      handleOptionSelect(option, byClick) {
+      handleOptionSelect(option) {
+        console.log('handleOptionSelect');
```
Comment (island205) [inline packages/select/src/select.vue]: :(

File: packages/select/src/option.vue [hunk 2]
```diff
@@ -2,6 +2,8 @@
   <li
     @mouseenter="hoverItem"
     @click.stop="selectOptionClick"
+    @mousedown="selectOptionMouseDown"
```
Comment (ziyoung) [inline packages/select/src/option.vue]: 有锅😅
测试用例：https://jsfiddle.net/a1L3vygb/9/
你会发现 blur 事件没触发。

**Comment by ziyoung** [comment]
related pr #11914

File: packages/select/src/option.vue (hunk 2 above)
Comment (island205) [reply packages/select/src/option.vue]: 我再看看

**Comment by wacky6** [comment]
multiple filterable 的时候，先选一个选项，然后点「x」移除这个选项，再点击输入框，触发了focus （预期不触发，因为Select依然是活动组件）

按照 commit 46ca1c （2017-11-07） 构建的fiddle： https://jsfiddle.net/6jLemnk9/

**Pull request closed without merging**